File: code/sorting/selection_sort/SelectionSort.php
<?php
// Part of Cosmos by OpenGenus Foundation

function find_min($arr){
	//traverse array
	for($i=0;$i<count($arr)-1;$i++){
		print(" index is ".$i."\n");
		if($arr[$i]>$arr[$i+1]){
			print("arr is ".$arr[$i]." arr2 is ".$arr[$i+1]."\n");
			$temp=$arr[$i+1];
			$arr[$i+1]=$arr[$i];
			$arr[$i]=$temp;
		}
	}
	return $arr;
}



$test = [1, 3, 43, 2, 14, 53, 12, 87, 33];

print_r($test);
function selection_sort($test){
	for($k=0;$k<count($test);$k++){
       		$test=find_min($test);
	}
	print_r($test);
}
selection_sort($test);

File: code/sorting/src/selection_sort/selection_sort.jl
# Part of Cosmos by OpenGenus Foundation

function selection_sort(array)
    for i in 1:length(array)
        minimum_index = i
        for j = i+1 : size(array)[1]
            if array[j] < array[minimum_index]
                minimum_index = j
            end
        end
        array[minimum_index], array[i] = array[i], array[minimum_index]
    end
    return array
end

testarray = [11,3,4,1,2,5,6,10]
selection_sort(testarray)